pkg/ipamd/ipamd.go

```diff
@@ -726,15 +769,22 @@ func (c *IPAMContext) tryAllocateENI() error {
 	return err
 }
 
-// For an ENI, try to fill in missing IPs on an existing ENI
-func (c *IPAMContext) tryAssignIPs() (increasedPool bool, err error) {
-	// If WARM_IP_TARGET is set, only proceed if we are short of target
-	short, _, warmIPTargetDefined := c.ipTargetState()
-	if warmIPTargetDefined && short == 0 {
+// For an ENI, try to fill in missing IPs on an existing ENI with PD disabled
+// try to fill in missing Prefixes on an existing ENI with PD enabled
+func (c *IPAMContext) tryAssignIPsOrPrefixes() (increasedPool bool, err error) {
```

Comment:
Same feedback as `tryUnassignIPsOrPrefixesFromAll`

Reply:
Same here, to avoid adding multiple conditions here - 

```
if c.isDatastorePoolTooLow() {
		c.increaseDatastorePool()
	} else if c.isDatastorePoolTooHigh() {
		c.decreaseDatastorePool(decreaseIPPoolInterval)
	}
```
I made a helper function which will tell based on the mode if the prefix pool is low/high or IP pool is low/high. Will take up the task of restructuring the code post this PR.